### ISSUE TITLE
linter: false positive with `unicorn/prefer-string-replace-all`

### ISSUE THREAD

**Issue by dhcmrlchtdj** [oxc-project/oxc#6259]
### Steps to Reproduce
Use the following code:
```javascript
// example.js
console.log("hello world".replaceAll(/hello|world/g, "replaced")) // log 'replaced replaced'
console.log("hello world".replaceAll("hello|world", "replaced")) // log 'hello world'
```

run with `oxlint -D=prefer-string-replace-all example.js` would throw error
```
  × eslint-plugin-unicorn(prefer-string-replace-all): This pattern can be replaced with `hello|world`.
   ╭─[example.js:1:38]
 1 │ console.log("hello world".replaceAll(/hello|world/g, "replaced")) // log 'replaced replaced'
   ·                                      ──────────────
 2 │ console.log("hello world".replaceAll("hello|world", "replaced")) // log 'hello world'
   ╰────
  help: Replace `/hello|world/g` with `"hello|world"`.
```

### Version
```
$ oxlint --version
Version: 0.9.9
```

### Investigation
I think this bug is caused by #5943 (cc @camchenry), as it only checks the first element of the regex pattern.
https://github.com/oxc-project/oxc/blob/3ef248afbd961195c223dbe5f5e0c73c970d40a1/crates/oxc_linter/src/rules/unicorn/prefer_string_replace_all.rs#L133
